src/components/OverviewCard.jsx

```diff
@@ -11,9 +23,114 @@ import {
 }) => {
+  const dropdownRef = useRef(null)
+  const fileMoveDropdownRef = useRef(null)
+  const categoryInputRef = useRef(null)
+  const [newCategory, setNewCategory] = useState('')
+  const [errorMessage, setErrorMessage] = useState('')
+  const [canShowDropdown, setCanShowDropdown] = useState(false)
+  const [canShowFileMoveDropdown, setCanShowFileMoveDropdown] = useState(false)
+  const [canShowDeleteWarningModal, setCanShowDeleteWarningModal] = useState(false)
+  const baseApiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}${category ? isResource ? `/resources/${category}` : `/collections/${category}` : ''}`
+
+  useEffect(() => {
+    if (canShowFileMoveDropdown) fileMoveDropdownRef.current.focus()
+    if (canShowDropdown) dropdownRef.current.focus()
+  }, [canShowFileMoveDropdown, canShowDropdown])
+
+  const moveFile = async (chosenCategory) => {
+    try {
+      // Retrieve data from existing page/resource
+      const resp = await axios.get(`${baseApiUrl}/pages/${fileName}`);
+
+      const { content, sha } = resp.data;
+      const base64DecodedContent = Base64.decode(content);
+      const { frontMatter, mdBody } = frontMatterParser(base64DecodedContent);
+      const {
+        title, permalink, file_url: fileUrl, date, third_nav_title: thirdNavTitle,
+      } = frontMatter;
+
+      const { editableLink } = retrieveCollectionAndLinkFromPermalink(permalink)
+
+      const fileInfo = {
+        title,
+        permalink: editableLink,
+        fileUrl,
+        date,
+        mdBody,
+        sha,
+        category: chosenCategory ? chosenCategory : newCategory,
+        prevCategory: category,
+        baseApiUrl,
+        type: isResource ? 'resource' : 'page',
+        thirdNavTitle,
+        fileName,
+        isNewFile: false,
+        siteName,
+      }
+      await saveFileAndRetrieveUrl(fileInfo)
+
+      // Refresh page
+      window.location.reload();
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  const deleteHandler = async () => {
+    try {
+      // Retrieve data from existing page/resource
+      const resp = await axios.get(`${baseApiUrl}/pages/${fileName}`);
+
+      const { sha } = resp.data;
+      const params = { sha };
+      await axios.delete(`${baseApiUrl}/pages/${fileName}`, {
+        data: params,
+      });
+
+      // Refresh page
+      window.location.reload();
+    } catch (err) {
+      console.log(err);
+    }
+  }
+  
+  const MenuItem = ({handler, id, children}) => {
+    return (
+      <div
+        id={id}
+        onClick={(e) => {
+          e.stopPropagation();
+          e.preventDefault(); 
+          if (handler) handler(e);
+        }}
+        className={`${elementStyles.dropdownItem}`}
+      >
+        {children}
+      </div>
+    )
+  }
+
+  const changeHandler = (event) => {
+    const { value } = event.target;
+
+    const errorMessage = validateCategoryName(value, isResource ? 'resource' : 'page')
```

Comment:
Also I think there's something wrong with the validation? We want to allow only alphanumeric + dash
<img width="355" alt="Screenshot 2020-11-18 at 7 35 37 PM" src="https://user-images.githubusercontent.com/31984694/99571292-09f6a580-2a0e-11eb-9dd3-a30a4b534a2b.png">